docs/source/tutorials/custom_environments.rst

```diff
@@ -0,0 +1,293 @@
+            # also allow it to equip the pickaxe
+            handlers.EquipAction(["diamond_pickaxe"])
+        ]
+
+Give Extra Observations
+====================================
+In addition to the POV image data the agent receives as an observation, lets provide
+it with compass and lifestats data. We override :code:`create_observables` just like the previous step.
+
+.. code-block:: python
+
+    def create_observables(self) -> List[Handler]:
+        return super().create_observables() + [
+            # current location and lifestats are returned as additional
+            # observations
+            handlers.ObservationFromCurrentLocation(),
+            handlers.ObservationFromLifeStats()
+        ]
+
+Set the Time 
+======================
+Lets set the time to morning.
+
+.. code-block:: python
+
+    def create_server_initial_conditions(self) -> List[Handler]:
+        return [
+            # Sets time to morning and stops passing of time
+            handlers.TimeInitialCondition(False, 23000)
+        ]
+
+Other Functions to Implement
+====================================
+
+:code:`SimpleEmbodimentEnvSpec` requires that we implement these methods.
+
+.. code-block:: python
+
+    # see API reference for use cases of these first two functions
+
+    def create_server_quit_producers(self):
+        return []
+    
+    def create_server_decorators(self) -> List[Handler]:
+        return []
+
+    # the episode can terminate when this is True
+    def determine_success_from_rewards(self, rewards: list) -> bool:
+        return sum(rewards) >= self.reward_threshold
+
+    def is_from_folder(self, folder: str) -> bool:
+        return folder == 'mlgwb'
+
+    def get_docstring(self):
+        return MLGWB_DOC
+
+**Congrats!** You just made your first MineRL environment. Checkout the herobraine API reference 
+to see many other ways to modify the world and agent.
+
+Using the Environment
+========================
+
+Now you need to solve it 🙂
+
+Create a new Jupyter Notebook (or Python) file in the same folder.
+
+In order to make an instance of the MLG Water Bucket Gym, we'll need to 
+register it with :code:`gym` then call :code:`gym.make`
+
+.. code-block:: python
+
+    import gym
+    from mlg_wb_specs import MLGWB
+
+    abs_MLG = MLGWB()
+    abs_MLG.register()
+    env = gym.make("MLGWB-v0")
+
+    # this line might take a couple minutes to run
+    obs  = env.reset()
+
+Here is some more boilerplate to get you started:
+
+.. code-block:: python
+
+    done = False
+
+    while not done:
+
+        env.render()
+
+        # a dictionary of actions. Try indexing it and changing values.
+        action = env.action_space.noop()
+
+        obs, reward, done, info = env.step(action)
+
+Running the previous two code blocks should open a Minecaft instance which 
+will quickly be minimized. Then, it should open a window that shows the agent's view.
+
+Try using a Jupyter Notebook and running the first code block
```

Comment:
Again, I would not talk about Jupyter notebooks. I would not encourage their use in the context of MineRL because of extra hassle of setting it up with no clear benefits (also harder to move code around)